app/models/post_mover.rb

```diff
@@ -176,6 +176,10 @@ def create_first_post(post)
 
     DiscourseEvent.trigger(:post_moved, new_post, original_topic.id)
 
+    # we don't want to keep the old topic's OP bookmarked when we are
+    # moving it into a new topic
+    Bookmark.where(post_id: post.id).update_all(post_id: new_post.id)
```

Comment:
This is necessary while the old query in `update_bookmarks` is not because if you move the OP of a topic into a new topic, the PostMover _duplicates_ the post into a new topic, leaving the original post behind in the original topic. This query moves the bookmark over to the new post.

We don't need to do any other updates for bookmarks in the post mover because the post ID of posts further down the topic do not change when moving to a new topic, and the bookmark table no longer has a topic ID to worry about.